it_serverless/serverless_test.py

```diff
@@ -0,0 +1,137 @@
+import collections
+import contextlib
+import json
+import os
+import subprocess
+import time
+import urllib.parse
+
+import pytest
+import requests
+from elasticsearch import Elasticsearch
+
+BASE_URL = os.environ["RALLY_IT_SERVERLESS_BASE_URL"]
+API_KEY = os.environ["RALLY_IT_SERVERLESS_API_KEY"]
+GET_CREDENTIALS_ENDPOINT = os.environ["RALLY_IT_SERVERLESS_GET_CREDENTIALS_ENDPOINT"]
+
+TRACKS = ["nyc_taxis", "pmc", "elastic/logs"]
+
+
+ServerlessProjectConfig = collections.namedtuple(
+    "ServerlessProjectConfig",
+    ["target_host", "username", "password"],
+)
+
+
+def serverless_api(method, endpoint, json=None):
+    resp = requests.request(
+        method,
+        BASE_URL + endpoint,
+        headers={
+            "Authorization": f"ApiKey {API_KEY}",
+            "Content-Type": "application/json",
+        },
+        json=json,
+        timeout=60,
+    )
+    resp.raise_for_status()
+    return resp.json()
+
+
+@pytest.fixture(scope="session")
+def serverless_project():
+    print("Creating project")
+    created_project = serverless_api(
+        "POST",
+        "/api/v1/serverless/projects/elasticsearch",
+        json={
+            "name": "rally-it-serverless",
+            "region_id": "aws-eu-west-1",
+        },
+    )
+
+    yield created_project
+
+    print("Deleting project")
+    serverless_api("DELETE", f"/api/v1/serverless/projects/elasticsearch/{created_project['id']}")
+
+
+@pytest.fixture(scope="session")
+def serverless_project_config(serverless_project):
+    credentials = serverless_api(
+        "POST",
+        f"/api/v1/serverless/projects/elasticsearch/{serverless_project['id']}{GET_CREDENTIALS_ENDPOINT}",
+    )
+
+    es_endpoint = serverless_project["endpoints"]["elasticsearch"]
+    es_hostname = urllib.parse.urlparse(es_endpoint).hostname
+    rally_target_host = f"{es_hostname}:443"
+
+    print("Waiting DNS propagation")
+    while True:
+        time.sleep(10)
+        with contextlib.suppress(subprocess.CalledProcessError):
+            subprocess.run(["nslookup", es_hostname, "8.8.8.8"], check=True)
+            break
+
+    print("Waiting for Elasticsearch")
+    while True:
+        try:
+            es = Elasticsearch(
+                f"https://{rally_target_host}",
+                basic_auth=(
+                    credentials["username"],
+                    credentials["password"],
+                ),
+                request_timeout=10,
+            )
+            info = es.info()
+            print("GET /")
+            print(json.dumps(info.body, indent=2))
+
+            authenticate = es.perform_request(method="GET", path="/_security/_authenticate")
+            print("GET /_security/_authenticate")
+            print(json.dumps(authenticate.body, indent=2))
+
+            break
+        except Exception as e:
+            print(f"GET / Failed with {type(e)}")
+            time.sleep(10)
```

Comment:
There's a potential of infinite wait here I think. Can we limit it time-wise?

Reply:
Thanks, done in [`80be169` (#1775)](https://github.com/elastic/rally/pull/1775/commits/80be169d84598f3d88a56e9c3b28406fefb11f51)